expression/builtin_miscellaneous.go

```diff
@@ -108,6 +109,27 @@ func (b *builtinSleepSig) Clone() builtinFunc {
 	return newSig
 }
 
+func sleepDuration(sessVars *variable.SessionVars, val float64) bool {
+	dur := time.Duration(val * float64(time.Second.Nanoseconds()))
+	ticker := time.NewTicker(10 * time.Millisecond)
+	defer ticker.Stop()
+	start := time.Now()
+	finish := false
+	for !finish {
+		select {
+		case now := <-ticker.C:
+			if now.Sub(start) > dur {
+				finish = true
+			}
+		default:
```

Comment:
I think it's a bug that we should not use `default` here, it checks `sessVars.Killed` even if the ticker does not fire.
It's better to check `sessVars.Killed` only when ticker fires.
Could you help fix?